lib/server_connection.ml

```diff
@@ -186,6 +186,8 @@ let set_error_and_handle ?request t error =
     t.error_handler ?request error (fun headers ->
       Writer.write_response writer (Response.create ~headers status);
       Body.of_faraday (Writer.faraday writer));
+    (* TODO(anmonteiro): uncomment to fix the failing tests. *)
+    (* wakeup_writer t; *)
```

Comment:
After some more investigation, I realized that the fix isn't as simple as I made it out to be, and I may not fully understand the problem. Calling `wakeup_writer` here does make the test pass but it doesn't work in an async scenario.